Collections/src/Track.cc

```diff
@@ -321,11 +322,81 @@ osu::Track::set_minDeltaRToElectrons (const edm::Handle<edm::View<TYPE(electrons
     dR = deltaR(*this, ele);
 
     if(dR < deltaRToClosestElectron_ || deltaRToClosestElectron_ < 0.0) deltaRToClosestElectron_ = dR;
-    if((*vidVetoMap)  [(*electrons).refAt(iEle)] && (dR < deltaRToClosestVetoElectron_   || deltaRToClosestVetoElectron_   < 0.0)) deltaRToClosestVetoElectron_   = dR;
-    if((*vidLooseMap) [(*electrons).refAt(iEle)] && (dR < deltaRToClosestLooseElectron_  || deltaRToClosestLooseElectron_  < 0.0)) deltaRToClosestLooseElectron_  = dR;
-    if((*vidMediumMap)[(*electrons).refAt(iEle)] && (dR < deltaRToClosestMediumElectron_ || deltaRToClosestMediumElectron_ < 0.0)) deltaRToClosestMediumElectron_ = dR;
-    if((*vidTightMap) [(*electrons).refAt(iEle)] && (dR < deltaRToClosestTightElectron_  || deltaRToClosestTightElectron_  < 0.0)) deltaRToClosestTightElectron_  = dR;
-  }
+
+    bool passesVeto_dxy   = false, passesVeto_dz   = false;
+    bool passesLoose_dxy  = false, passesLoose_dz  = false;
+    bool passesMedium_dxy = false, passesMedium_dz = false;
+    bool passesTight_dxy  = false, passesTight_dz  = false;
+
+    // Note in below, these remain false if |eta| >= 2.5; thus an eta cut is also being applied here as intended
+#if CMSSW_VERSION_CODE < CMSSW_VERSION(9,4,0)
```

Comment:
L332-370: Why are all these dxy cut values hardcoded?

Reply:
The EGM POG doesn't include vertex requirements in the VIDs for various reasons, so these cuts need to be made explicitly (they were forgotten before this PR by mistake). Plus this isn't the normal framework "electrons" collection, so they can't be provided in the typical cut PSet fashion.

Are you suggesting however that these be external PSet parameters?

Reply:
Ah ok, thanks for the information. But yeah, wouldn't external PSet parameters be a good idea?

Reply:
I can do that; it'll look cleaner too.